### PR TITLE
Change of error handling

### PR DESCRIPTION
**Pull request by rakaar**
Previously error handling code was containing too much amount of characters - i meant 2 lines one to write header and other to write message and the long loggging line

it was disturbing 
anyone who wants to debug would feel like reading code between error handling not err handling btn code
hence i changed it to
small and simpler lines 
```
utils.LOG.Println(err)
http.Error(w, err.Error(), 400)
```
where where LOG in pacakge utils is also one line of log variable
```
var LOG = log.New(os.Stderr, "Error: ", log.LstdFlags | log.Lshortfile)
```

with too many long lines of code, it will be unreadable and difficult to debug in future
here in doing so, i m making few things wrongly
1. instead of w.Write which is used to return success, i m using http.Error , hence violating uniformity:
i feel we should replace w.write in replacing if possible, because in most of cases(atleast in whole of kwoc) i see we are returning 200 status only, then y are we even mentioning it everytime? if there is any replacement then lets do it
2. Levels of logging are being ignored
For seperate levels we can use some libs like logrus and all, but is it even necessary, like what do we want to log at different levels, I  see mostly the need of logging at errror level only and nothing else except at beginning of starting of server, which is logging at info level, which can done through regular fmt too ! or logged seperately without using the logger of utils.

i m not concerned about standard practices. but i feel that it is important to have a code that is readable to most of others and can be changed or debugged easily when required and easy to write. keeping these in mind only , i want to change the code, if it is okay then i will change in other controllers too.

### CODE REVIEW

**Comment by yashrsharma44** [comment]
> i m not concerned about standard practices.

I would differ from this opinion, because if we have a custom error practices, it would become difficult to check what is the issue, rather than that, the previous code sounds fair, as if there are bugs in it, we can google search the issue, and more often than not, as it is a standard issue, we would certainly have a fix. Not possible, if we have custom handling of errors

**Comment by rakaar** [comment]
@yashrsharma44 
you took these lines `i m not concerned about standard practices.`  very seriously, lol. 
so currently, I m not using any custom functions. I have not written anything on my own.
it is a logger being used from standard go package "logs", defined in utils with a path defined(same as prev logger) and with flags set so as to also print the line number.(same as the prev logger) 
And that is being imported in every controller to log, the reason to do so that i don't have to define the path and flags of logger every time 
Also,  standard go package "logs" will have more proper documentation and support than any library.
I have referred from docs itself.

**Comment by yashrsharma44** [comment]
Well, if you are using an alias for a behaviour, like `LOG` , then make sure that we don't use it for common logging, because the api is non-regular, and that's my only concern.
Someone else might just use it for logging, and that would create unnecessary confusion.

I do agree that the code becomes a bit lengthy, but it is a part of standard practice, so if we refer documentation, we can get explanation of the api. 

To think of an analogy, in golang we have a common pattern of checking error and logging / raising an error. To combat this repititive behaviour, we can create a function and then abstract the repititive behaviour. But usually we don't do that because it's just not the part of the coding convention for Golang.

But, if we can get the ` LOG` function documented and how should it be used, then we can move forward with this approach.

**Comment by rakaar** [comment]
> But, if we can get the LOG function documented and how should it be used, then we can move forward with this approach

Glad you raise the point regarding documentation. I was also thinking about docs of endpoints too. 
So for now, where shall we document things like logging? -README.md or a separate file - docs.md

**Comment by rakaar** [comment]
Regarding that line of standard practices, i wished to address this point

In case of error, we are doing w.WriteHeader and one more line w.Write, the same thing can be done using http.Error() in simpler way
But the reason we have chose w.Write is that we thought uniformity is a good `standard` because we are using w.Write on returning response messages also, but never thought this would increase the code. 
What is your opinion on returning response? like how do u want it.
Are u okay with http.Error() in err handling ?
Any other ideas for w.Write alternatives for returning success response, which would match with http.Error to avoid the  w.WriteHeader(200)  everytime

**Comment by yashrsharma44** [comment]
> So for now, where shall we document things like logging? -README.md or a separate file - docs.md

I would suggest to document a functions behaviour in the code itself, using comments

For describing the flow of code, or logical meaning of different components, we can create a doc.md to explain the same

**Comment by yashrsharma44** [comment]
> What is your opinion on returning response? like how do u want it.

Umm I don't have a preference as such, but just make sure that we use similar logic while we write response 😉

**Comment by rakaar** [comment]
indirectly are u suggesting using w.writeHeader and w.Write in both places , err handling and response?

What is wrong in 
http.Error() being used in case of error([suggested as per go docs](https://golang.org/doc/articles/wiki/#tmp_9)) and w.Write in case of success responses

**Comment by rakaar** [comment]
@sudo-rgorai

**Comment by yashrsharma44** [comment]
> indirectly are u suggesting using w.writeHeader and w.Write in both places , err handling and response?

What is wrong in
http.Error() being used in case of error(suggested as per go docs) and w.Write in case of success responses

I think we can use `http.Error()` for returning error response, no issues with that. Just avoid using some non-standard practice. The above looks fine though :+1:

File: docs.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+### Logging
```
Comment (yashrsharma44) [inline docs.md:1]: Good Addition!

**Pull request merged**